### PR TITLE
Feat(coordinator): Extract check-in/out times from event description

### PR DESCRIPTION
**Pull request by tykeal**
## Summary

Enhances the `honor_event_times` feature to extract check-in/check-out times from event descriptions when calendar events are all-day (VALUE=DATE). This supports PMS systems like Hostaway that embed times in the DESCRIPTION field rather than using datetime DTSTART/DTEND.

## Time Resolution Priority (Updated)

When `honor_event_times=True`:
1. Explicit DTSTART/DTEND datetime → use event times (existing)
2. **NEW**: All-day event + description contains check-in/out times → use extracted times
3. Override exists (fallback for partially missing description times) → use override times
4. Fallback → configured CONF_CHECKIN/CONF_CHECKOUT defaults

Description-extracted times take priority over overrides. If only one time (check-in or check-out) is found in the description, the other falls through to override or default.

## Supported Formats

- Integer hours: `Checkin time: 16` → 16:00
- HH:MM: `Check-in time: 16:30` → 16:30
- 12-hour: `Check-in: 4 PM` → 16:00
- Case-insensitive, multiple pattern variants
- Boundary-guarded: rejects malformed values like `16:300`

## New Files

- `custom_components/rental_control/description_parser.py` — Pure-function time extraction module
- `tests/unit/test_description_parser.py` — Comprehensive unit tests (all formats + edge cases)

## Test Results

All 710 tests pass, no regressions.

Closes #482

### CODE REVIEW

File: custom_components/rental_control/description_parser.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# SPDX-FileCopyrightText: 2025 Andrew Grimberg <tykeal@bardicgrove.org>
+# SPDX-License-Identifier: Apache-2.0
+"""Parse check-in/check-out times from iCal event descriptions."""
+
+from __future__ import annotations
+
+from datetime import time
+import re
```
[the remark this answers is not in the document]
Comment (tykeal) [reply custom_components/rental_control/description_parser.py:8]: Fixed in d54ca24 — reordered imports (ruff auto-fixed on commit).

File: custom_components/rental_control/description_parser.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+_CHECKIN_PATTERN = re.compile(
+    r"check-?in(?:\s+time)?\s*:\s*(\d{1,2})(?::(\d{2}))?\s*(AM|PM)?",
+    re.IGNORECASE,
+)
+
+_CHECKOUT_PATTERN = re.compile(
+    r"check-?out(?:\s+time)?\s*:\s*(\d{1,2})(?::(\d{2}))?\s*(AM|PM)?",
```
[the remark this answers is not in the document]
Comment (tykeal) [reply custom_components/rental_control/description_parser.py]: Fixed in d54ca24 — same boundary fix applied to the checkout pattern.

File: custom_components/rental_control/coordinator.py [hunk 3]
```diff
@@ -622,6 +624,36 @@ async def _ical_parser(
                     # FR-003: PMS times take priority for timed events
                     checkin: time = event["DTSTART"].dt.time()
                     checkout: time = event["DTEND"].dt.time()
+                elif self.honor_event_times and not has_explicit_times:
+                    # Priority 2: description-extracted times (NEW)
+                    description = str(event.get("DESCRIPTION", ""))
```
[the remark this answers is not in the document]
Comment (tykeal) [reply custom_components/rental_control/coordinator.py]: Fixed in d54ca24 — now uses `raw_desc = event.get("DESCRIPTION")` with `str(raw_desc) if raw_desc else ""`.

File: custom_components/rental_control/description_parser.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tykeal) [reply custom_components/rental_control/description_parser.py]: Fixed in d54ca24 — restructured the regex to use `(\d{1,2}(?::\d{2})?)(?!\d|:)` which captures the full time token with a negative lookahead preventing partial matches. `"16:300"` now correctly returns None.

File: custom_components/rental_control/description_parser.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
+# SPDX-FileCopyrightText: 2025 Andrew Grimberg <tykeal@bardicgrove.org>
+# SPDX-License-Identifier: Apache-2.0
+"""Parse check-in/check-out times from iCal event descriptions."""
+
+from __future__ import annotations
+
+from datetime import time
+import re
```
[the remark this answers is not in the document]
Comment (tykeal) [reply custom_components/rental_control/description_parser.py:8]: This is a false positive — `ruff check --select I` passes cleanly. The project's isort config (`force-sort-within-sections = true`) interleaves imports alphabetically by module name, so `from datetime` correctly precedes `import re` (`d` < `r`).

File: custom_components/rental_control/coordinator.py [hunk 5]
```diff
@@ -622,6 +624,37 @@ async def _ical_parser(
...
+                elif self.honor_event_times and not has_explicit_times:
+                    # Priority 2: description-extracted times (NEW)
+                    raw_desc = event.get("DESCRIPTION")
+                    description = str(raw_desc) if raw_desc else ""
+                    desc_checkin = extract_checkin_time(description)
+                    desc_checkout = extract_checkout_time(description)
+
+                    if override:
+                        # Priority 3 fallback for missing description times
+                        start_tz = override["start_time"].astimezone(self.timezone)
```
[the remark this answers is not in the document]
Comment (tykeal) [reply custom_components/rental_control/coordinator.py:636]: Good catch on the PR description wording. The implementation is correct per the spec: description-extracted times have higher priority than overrides (Priority 2 > Priority 3). Updated the PR description to clarify the actual priority chain:

1. Explicit datetime DTSTART/DTEND → use event times
2. All-day + description times → use extracted times
3. Override (for missing description values) → use override times
4. Fallback → configured defaults
